Some editor config updates that at least Visual Studio uses:

alextd opened the pull request:
**Code Style**

## Summary

a few `.editorconfig` settings that Visual Studio will use to enforce the code style:
- New lines for control blocks
- Align the pointer `*` to the right
- No space after if

(More could be done but these trip me up all the time)
Possible values are here:
https://learn.microsoft.com/en-us/visualstudio/ide/cpp-editorconfig-properties?view=vs-2022
Global settings are in Tools -> Options -> Text Editor -> C++ -> Code Style -> Formatting -> (various) which are overridden by the value in .editorconfig

(I dunno where the actual code style rules are and if they exist, why doesn't visual studio know about them already so I have to duplicate them here)

TomGoodIdea commented on .editorconfig:
```diff
@@ -14,6 +14,10 @@ indent_style = tab
 # Code files
 [{*.{cpp,h,rc,hpp}, SConstruct, SConscript}]
 indent_style = tab
+trim_trailing_whitespace = true
```
How does this work with do..while loops? Our current convention is to place the braces on a new line only after a `)`.

TomGoodIdea commented on .editorconfig:
```diff
@@ -14,6 +14,10 @@ indent_style = tab
 [{*.{cpp,h,rc,hpp}, SConstruct, SConscript}]
 indent_style = tab
+trim_trailing_whitespace = true
+cpp_new_line_before_open_brace_block = new_line
```
How does this work with do..while loops? Our current convention is to place the braces on a new line only after a `)`.

alextd commented:
Yup, that would apply to `do` blocks as well so I guess we can't have that setting here, removed it.

Also removed the trailing whitespace as I noticed it was already in [*]. (I assumed it wasn't being formatted since I always have this problem, but it's just that Visual Studio needs you to actually trigger the formatting. It won't stop you from actually making the mistakes)

tibetiroka commented on .editorconfig:
```diff
@@ -14,6 +14,9 @@ indent_style = tab
 # Code files
 [{*.{cpp,h,rc,hpp}, SConstruct, SConscript}]
 indent_style = tab
+#cpp_new_line_before_open_brace_block = new_line # also applies to do {} so can't have this
```
```suggestion
```
